lib/auth/user.go

```diff
@@ -46,15 +55,22 @@ func (s *AuthServer) CreateUser(ctx context.Context, user services.User) error {
 		connectorName = user.GetCreatedBy().Connector.ID
 	}
 
-	s.EmitAuditEvent(events.UserCreate, events.EventFields{
-		events.EventUser:       user.GetCreatedBy().User.Name,
-		events.UserExpires:     user.Expiry(),
-		events.UserRoles:       user.GetRoles(),
-		events.UserCreatedName: user.GetName(),
-		events.UserConnector:   connectorName,
-	})
+	event := events.EventFields{
+		events.UserExpires:   user.Expiry(),
+		events.UserRoles:     user.GetRoles(),
+		events.UserConnector: connectorName,
+		events.EventUser:     updateBy,
+	}
 
-	return nil
+	if userExisted {
+		event[events.UserUpdatedName] = user.GetName()
+		s.EmitAuditEvent(events.UserUpdate, event)
```

Comment:
Instead of having `created_user` vs `updated_user` we can have `entity` field to capture this. 

```
code: T1004I|T1002I
entity: 'mama'
user: 'system'
```

Note that event and code fields tell us what type of operation it is (update/create/delete/etc). @benarent  could you confirm the proposed structure for these events.

Reply:
revised to `entity` field. @benarent let me know if you want me to change it to something different.